### ISSUE TITLE
Flaky test: getPermissions_updatedWithNewRealms_stressTest

### ISSUE THREAD

**Issue by rorbech** [realm/realm-java#7086]
From https://ci.realm.io/blue/organizations/jenkins/realm%2Frealm-java/detail/PR-7053/6/tests
```
09-03 15:30:08.476 D/REALM_SYNC( 3461): Connection[5]: Session[5]: Activating
09-03 15:30:08.476 I/REALM_SYNC( 3461): Connection[5]: Session[5]: client_reset_config = true, Realm exists = true, async open = false, client reset = true
09-03 15:30:08.476 I/REALM_SYNC( 3461): Connection[5]: Session[5]: Client reset config, metadata_dir = '/data/data/io.realm.test/files/realm-object-server/603c02f77919b177343b8c78cb7f6a85/603c02f77919b177343b8c78cb7f6a85/test4.resync', recover_local_changes = true, require_recent_state_realm = true
09-03 15:30:08.477 D/REALM_SYNC( 3461): Connection[5]: Session[5]: Create ClientStateDownload, realm_path = /data/data/io.realm.test/files/realm-object-server/603c02f77919b177343b8c78cb7f6a85/603c02f77919b177343b8c78cb7f6a85/test4, metadata_dir = /data/data/io.realm.test/files/realm-object-server/603c02f77919b177343b8c78cb7f6a85/603c02f77919b177343b8c78cb7f6a85/test4.resync
09-03 15:30:08.477 D/REALM_SYNC( 3461): Connection[5]: Session[5]: ClientStateDownload: initialize_from_new using directory, m_versioned_meta_data_dir = '/data/data/io.realm.test/files/realm-object-server/603c02f77919b177343b8c78cb7f6a85/603c02f77919b177343b8c78cb7f6a85/test4.resync/version-1'
09-03 15:30:08.599 W/adbd    ( 3303): timeout expired while flushing socket, closing
09-03 15:30:08.668 D/REALM_SYNC( 3461): Connection[5]: Session[5]: client_file_ident = 0, client_file_ident_salt = 0
09-03 15:30:08.668 D/REALM_SYNC( 3461): Opening Realm file: /data/data/io.realm.test/files/realm-object-server/603c02f77919b177343b8c78cb7f6a85/603c02f77919b177343b8c78cb7f6a85/test7
09-03 15:30:08.687 D/REALM_SYNC( 3461): Connection[4]: Session[4]: Progress handler called, downloaded = 0, downloadable(total) = 0, uploaded = 0, uploadable = 473, reliable_download_progress = 0, snapshot version = 2
09-03 15:30:08.687 D/REALM_SYNC( 3461): Opening Realm file: /data/data/io.realm.test/files/realm-object-server/603c02f77919b177343b8c78cb7f6a85/603c02f77919b177343b8c78cb7f6a85/test4
09-03 15:30:08.687 D/REALM_SYNC( 3461): Connection[5]: Session[5]: Progress handler called, downloaded = 0, downloadable(total) = 0, uploaded = 0, uploadable = 473, reliable_download_progress = 0, snapshot version = 2
09-03 15:30:08.687 D/REALM_SYNC( 3461): Connection[4]: Session[4]: Initiating deactivation
09-03 15:30:08.688 D/REALM_SYNC( 3461): Connection[4]: Session[4]: Deactivation completed
09-03 15:30:08.786 I/TestRunner( 3461): failed: getPermissions_updatedWithNewRealms_stressTest(io.realm.PathLevelPermissionsTests)
09-03 15:30:08.787 I/TestRunner( 3461): ----- begin exception -----
09-03 15:30:08.787 I/TestRunner( 3461): java.lang.AssertionError
09-03 15:30:08.787 I/TestRunner( 3461): 	at org.junit.Assert.fail(Assert.java:86)
09-03 15:30:08.787 I/TestRunner( 3461): 	at org.junit.Assert.assertTrue(Assert.java:41)
09-03 15:30:08.787 I/TestRunner( 3461): 	at org.junit.Assert.assertTrue(Assert.java:52)
09-03 15:30:08.787 I/TestRunner( 3461): 	at io.realm.PathLevelPermissionsTests.getPermissions_updatedWithNewRealms_stressTest(PathLevelPermissionsTests.java:140)
09-03 15:30:08.787 I/TestRunner( 3461): 	at java.lang.reflect.Method.invoke(Native Method)
09-03 15:30:08.787 I/TestRunner( 3461): 	at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:50)
09-03 15:30:08.787 I/TestRunner( 3461): 	at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:12)
09-03 15:30:08.787 I/TestRunner( 3461): 	at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:47)
09-03 15:30:08.787 I/TestRunner( 3461): 	at org.junit.internal.runners.statements.InvokeMethod.evaluate(InvokeMethod.java:17)
09-03 15:30:08.787 I/TestRunner( 3461): 	at android.support.test.internal.runner.junit4.statement.RunBefores.evaluate(RunBefores.java:80)
09-03 15:30:08.787 I/TestRunner( 3461): 	at android.support.test.internal.runner.junit4.statement.RunAfters.evaluate(RunAfters.java:61)
09-03 15:30:08.787 I/TestRunner( 3461): 	at io.realm.rule.RunInLooperThread$TestThread.run(RunInLooperThread.java:529)
09-03 15:30:08.787 I/TestRunner( 3461): 	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:462)
09-03 15:30:08.787 I/TestRunner( 3461): 	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
09-03 15:30:08.787 I/TestRunner( 3461): 	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1167)
09-03 15:30:08.787 I/TestRunner( 3461): 	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:641)
09-03 15:30:08.787 I/TestRunner( 3461): 	at java.lang.Thread.run(Thread.java:919)
09-03 15:30:08.787 I/TestRunner( 3461): ----- end exception -----
09-03 15:30:08.788 I/TestRunner( 3461): finished: getPermissions_updatedWithNewRealms_stressTest(io.realm.PathLevelPermissionsTests)
```

**Comment by edualonso** [comment]
This test is only present in the master branch but it has been removed in the v10 release, so should not be a concern once v10 is merged into master.
